equinox/nn/attention.py

```diff
@@ -219,22 +254,14 @@ def __call__(
         key_heads = self._project(self.key_proj, key_)
         value_heads = self._project(self.value_proj, value)
 
-        logits = jnp.einsum("shd,Shd->hsS", query_heads, key_heads)
-        logits = logits / math.sqrt(self.qk_size)
-        if mask is not None:
-            if mask.shape != logits.shape:
-                raise ValueError(
-                    f"mask must have shape (num_heads, query_seq_length, "
-                    f"kv_seq_length)=({self.num_heads}, {query_seq_length}, "
-                    f"{kv_seq_length}). Got {mask.shape}."
-                )
-            logits = jnp.where(mask, logits, -jnp.inf)
-
-        weights = jax.nn.softmax(logits, axis=-1)
-        weights = self.dropout(
+        dropout_fn = lambda weights: self.dropout(
             weights, key=key, inference=inference, deterministic=deterministic
         )
-        attn = jnp.einsum("hsS,Shd->shd", weights, value_heads)
+        # vmap over attention heads
+        in_axes = (1, 1, 1, None if mask is None else 1, None)
```

Comment:
I think we need to specify `out_axes` as 1 as well. Otherwise I think the output of the `MultiheadAttention` layer will be different as compared to the previous version of Equinox.